### PR TITLE
Add `brightness` to CupertinoNavigationBar (fixes #46216)

### LINKED ISSUE #46216: Hope that CupertinoNavigationBar provides a 'brightness'  property like AppBar
as title

### PR DESCRIPTION
**Pull request by KagurazakaHanabi**
## Related Issues

Closes #46216

## Tests

I added the following tests:

- Can specify custom brightness

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by LongCatIsLooong — COMMENTED** [review]
Thank you for the PR! Looks great overall.

File: packages/flutter/test/cupertino/nav_bar_test.dart [hunk 1]
```diff
@@ -136,6 +136,68 @@ void main() {
...
+        Widget widget) => widget is AnnotatedRegion);
+    expect(region2.value, SystemUiOverlayStyle.dark);
+
+    await tester.pumpWidget(
+      const CupertinoApp(
+        home: CustomScrollView(
+          slivers: <Widget>[
+            CupertinoSliverNavigationBar(
+              backgroundColor: Color(0xF0F9F9F9),
+              brightness: Brightness.dark,
```
Comment (LongCatIsLooong) [inline packages/flutter/test/cupertino/nav_bar_test.dart]: CI is complaining `Failed assertion: line 1231 pos 7: 'largeTitleContent != null'`

File: packages/flutter/test/cupertino/nav_bar_test.dart [hunk 2]
```diff
@@ -136,6 +136,68 @@ void main() {
...
+    await tester.pumpWidget(
+      const CupertinoApp(
+        home: CupertinoNavigationBar(
+          backgroundColor: Color(0xF0F9F9F9),
+          brightness: Brightness.dark,
+        ),
+      ),
+    );
+
+    final AnnotatedRegion<SystemUiOverlayStyle> region1 = tester.allWidgets.singleWhere((
```
[the remark this answers is not in the document]
Comment (KagurazakaHanabi) [reply packages/flutter/test/cupertino/nav_bar_test.dart]: It will cause `Bad state: No element`

[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/test/cupertino/nav_bar_test.dart]: Ah you were right `AnnotatedRegion` has a type parameter. Sorry ignore the suggestion.

[the remark this answers is not in the document]
Comment (KagurazakaHanabi) [reply packages/flutter/test/cupertino/nav_bar_test.dart]: `tester.widget<AnnotatedRegion<SystemUiOverlayStyle>>(find.byType(AnnotatedRegion));` also cause this error

[the remark this answers is not in the document]
Comment (KagurazakaHanabi) [reply packages/flutter/test/cupertino/nav_bar_test.dart]: `tester.widget<AnnotatedRegion<SystemUiOverlayStyle>>(find.byType(AnnotatedRegion));` also cause this error

[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/test/cupertino/nav_bar_test.dart]: Unfortunately there's https://github.com/dart-lang/sdk/issues/33297. IMO using `allWidgets.singleWhere` is fine.

**Review by LongCatIsLooong — APPROVED** [review]

**Comment by LongCatIsLooong** [comment]
@KagurazakaHanabi unfortunately the commit had to be reverted due to the following failure (https://github.com/flutter/flutter/runs/363694582):

```
  error • A value of type 'Widget' can't be assigned to a variable of type 'AnnotatedRegion<SystemUiOverlayStyle>' • packages/flutter/test/cupertino/nav_bar_test.dart:149:59 • invalid_assignment
  error • A value of type 'Widget' can't be assigned to a variable of type 'AnnotatedRegion<SystemUiOverlayStyle>' • packages/flutter/test/cupertino/nav_bar_test.dart:162:59 • invalid_assignment
  error • A value of type 'Widget' can't be assigned to a variable of type 'AnnotatedRegion<SystemUiOverlayStyle>' • packages/flutter/test/cupertino/nav_bar_test.dart:180:59 • invalid_assignment
  error • A value of type 'Widget' can't be assigned to a variable of type 'AnnotatedRegion<SystemUiOverlayStyle>' • packages/flutter/test/cupertino/nav_bar_test.dart:198:59 • invalid_assignment
```

This is presumably because implicit cast is recently prohibited. Could you try updating the PR branch and casting the expression to `AnnotatedRegion<SystemUiOverlayStyle>` before assigning it to the variable?

**Comment by KagurazakaHanabi** [comment]
@LongCatIsLooong I fixed it

**Comment by LongCatIsLooong** [comment]
Awesome thank you! Could you open a new PR for it?

**Pull request merged**